libsolidity/analysis/TypeChecker.cpp

```diff
@@ -608,6 +608,20 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 		}
 	}
 
+	auto fixedPoint = dynamic_cast<FixedPointType const*>(_variable.annotation().type);
+	if (fixedPoint)
+	{
+		bigint fixedPointTypeSize = boost::multiprecision::pow(bigint(2), fixedPoint->numBits());
+		bigint digitSize = boost::multiprecision::pow(bigint(10), fixedPoint->fractionalDigits());
+		if (fixedPointTypeSize <= digitSize)
+			m_errorReporter.typeError(
+				5108_error,
+				_variable.location(),
+				"Invalid fixed point type " + fixedPoint->toString(true) + ": 10^"
+				+ std::to_string(fixedPoint->fractionalDigits()) + " does not fit in 2^"
+				+ std::to_string(fixedPoint->numBits()) + " bits."
```

Comment:
I think that this message might be a bit too cryptic for users. It might not be clear what the compiler is trying to accomplish by fitting `10^N` into `2^M` bits. I think that the message should say simply that it's impossible to express 1.0 in the type.

How about this?
```
Invalid fixed-point type: ufixed2x5. The fixed-point representation of 1.0 with 5 fractional decimal digits requires 17 bits but the type provides only 2.
```

Reply:
Ah nice! Maybe it's really better to put it like this. I agree that it is easier to understand.